src/main/java/net/twasi/obsremotejava/OBSCommunicator.java

```diff
@@ -164,19 +171,19 @@ public void onConnect(Session session) {
             fut = session.getRemote().sendStringByFuture(new Gson().toJson(new GetVersionRequest(this)));
             fut.get(2, TimeUnit.SECONDS);
         } catch (Throwable t) {
-            t.printStackTrace();
+            runOnError("An error occurred while trying to get a session", t);
         }
     }
 
     @OnWebSocketMessage
     public void onMessage(String msg) {
         if (msg == null) {
-            System.out.println("Ignored empty message");
+            log.debug("Ignored empty message");
             return;
         }
 
         if (debug) {
```

Comment:
I think since we use proper logging now, we could remove this debug mode / debug flag here and just print it out to the debug (or here trace would match better in my opinion).

Reply:
Agreed, great point -- if fact, for computation-intensive stuff we could use log.isDebugEnabled or similar to wrap expensive logging operations. Might revisit this later as another PR.